ykrt/pass_finder/cargo_run.py

```diff
@@ -0,0 +1,59 @@
+import os
+import subprocess
+import time
+
+def run_test(yk_path, env=None, n=1):
+    # Set path and environment variables
+    yk_path = os.environ.get('YK_PATH')
+    if yk_path is None:
+        raise ValueError("YK_PATH environment variable is not set")
+
+    yklua_path = os.environ.get('YKLUA_PATH')
+    if yklua_path is None:
+        raise ValueError("YKLUA_PATH environment variable is not set")
+
+    if env is None:
+        env = os.environ.copy()
+
+    env['PATH'] = f"{yk_path}/bin:{env['PATH']}"
+    env['YK_BUILD_TYPE'] = 'debug'
+     
+    # Check if pre and post-link flags are set
+    prelink_passes = (env or os.environ).get('PRELINK_PASSES', 'Not Set')
+    print(f"\n\033[93mPRELINK_PASSES: {prelink_passes}\033[0m")
+    postlink_passes = (env or os.environ).get('POSTLINK_PASSES', 'Not Set')
+    print(f"\n\033[93mPOSTLINK_PASSES: {postlink_passes}\033[0m")
+ 
+    curdir = os.getcwd()
+    os.chdir(yklua_path)
+
+    times = []
+    c_test = None 
+
+    for _ in range(n):
+        subprocess.run(["make clean"], shell=True, env=env)
+        c = subprocess.run(["make && timeout 30 sh test.sh"], shell=True, env=env or os.environ)
+        print(f"\033[95m returncode for c is {c.returncode} \033[0m\n")
```

Comment:
In general I'm allergic to chatty scripts as we tend to end up not being able to see the forest for the trees. I also suspect that we should be checking `c.returncode` here before bothering with `cargo test`? Running the latter if the former failed seems pointless?

Reply:
If Iti finds these messages useful for debugging and development, perhaps there should be a verbose flag (off by default)?

Reply:
Addressed here: 6c3f62f2dd481f0a792a6e681a1d4f0fe6b336fc